### ISSUE TITLE
OCATS :-  we are stuck up in deploying ear and war files. 

### ISSUE THREAD

**Issue by Samruddhi123** [Comcast/cats#29]
Query as mentioned below:- 
--------------------------------------------------------------------------------------------------------
We are able to install the OCATS UI but we are stuck up in deploying ear and war files. 
We are still troubleshooting the same but cannot conclude on the same. 

•	How to resolve the issue of deployment of ear & war files? 
•	Is this due a hardware dependency / any other reason? 

Here are the snapshots attached:-
1.	Screenshot 1  status  not being enabled
2.	Screenshot 2 ear file error  internal server error
3.	Screenshot 3  ear file error  internal server error

PFA:- 
![screenshot 1](https://cloud.githubusercontent.com/assets/20223460/16484839/3169f84e-3ed9-11e6-8ee0-2bf39883efde.png)
![screenshot 2](https://cloud.githubusercontent.com/assets/20223460/16484838/3160853e-3ed9-11e6-86e2-1198e3569ecf.png)
![screenshot 3](https://cloud.githubusercontent.com/assets/20223460/16484837/315e0e94-3ed9-11e6-8e65-7012ebf70f39.png)

**Comment by sajayjk** [comment]
The above screenshots are not useful. Can you please attach the Jboss server.log file that has the errors regarding the deployment.

**Comment by XsubinX** [comment]
Missing data source.  Is data sources configured correctly ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA

**Comment by Samruddhi123** [comment]
How to configure the data sources  === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA

**Comment by Samruddhi123** [comment]
How to configure the data sources === > Can u please guide us ?

Here are the Jboss server.log file attached :- PFA


[logs.txt](https://github.com/Comcast/cats/files/351456/logs.txt)

**Comment by XsubinX** [comment]
Refer

- https://developer.jboss.org/wiki/DataSourceConfigurationInAS7
- https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc
- https://github.com/Comcast/cats/blob/master/services/video-recorder/war/src/main/resources/sql/model/mysql/pvr_schema.sql
- https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/key_manager.sql
- https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/persistence.xml

**Comment by LTComcast** [comment]
Please find the attached logs from Chennai team. 
Note: We have referred the above links to create databases and persistence file.

Still we are facing the same issue while deploying the files.
[Logs_Chennai.zip](https://github.com/Comcast/cats/files/351745/Logs_Chennai.zip)

**Comment by Samruddhi123** [comment]
Thanks for the extended help.
# #_**We tried all the steps given above given by Subin Sugunan . bUt still facing the same issue ... logs already attached above.**_

•https://developer.jboss.org/wiki/DataSourceConfigurationInAS7
•https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc
•https://github.com/Comcast/cats/blob/master/services/video-recorder/war/src/main/resources/sql/model/mysql/pvr_schema.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/key_manager.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/persistence.xml

**Comment by Samruddhi123** [comment]
Thanks for the extended help.

### **We tried all the steps given above given by Subin Sugunan . bUt still facing the same issue ... logs already attached above.** 

•https://developer.jboss.org/wiki/DataSourceConfigurationInAS7
•https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc
•https://github.com/Comcast/cats/blob/master/services/video-recorder/war/src/main/resources/sql/model/mysql/pvr_schema.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/key_manager.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/persistence.xml

**Comment by Samruddhi123** [comment]
Thanks for the extended help.

**We tried all the steps given above given by Subin Sugunan . bUt still facing the same issue ... logs already attached above.** 

•https://developer.jboss.org/wiki/DataSourceConfigurationInAS7
•https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc
•https://github.com/Comcast/cats/blob/master/services/video-recorder/war/src/main/resources/sql/model/mysql/pvr_schema.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/key_manager.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/persistence.xml

**Comment by Samruddhi123** [comment]
Thanks for the extended help.

**#We tried all the steps given above given by Subin Sugunan . But still facing the same issue ... logs already attached above.** 

•https://developer.jboss.org/wiki/DataSourceConfigurationInAS7
•https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc
•https://github.com/Comcast/cats/blob/master/services/video-recorder/war/src/main/resources/sql/model/mysql/pvr_schema.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/key_manager.sql
•https://github.com/Comcast/cats/blob/master/services/ir/keymanager/src/main/resources/META-INF/persistence.xml

**Comment by cfrede001** [comment]
Please provide the server information:
OS/Platform Centos/Ubuntu/Debian/etc and version
Java version
Mysql version
The standalone.xml file from JBoss configuration.

**Comment by Samruddhi123** [comment]
Hello,

1. OS we are using is CentOS 7 ==> centos-release-7-2.1511.el7.centos.2.10.x86_64
2. Java version "1.7.0_80"
    Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
    Java HotSpot(TM) 64-Bit Server VM (build 24.80-b11, mixed mode)
3. MySQL connection id is 2
    Server version: 5.6.31 MySQL Community Server (GPL)
4. Attached both the files ==> standalone.xml and standalone-full.xml


[standalone.txt](https://github.com/Comcast/cats/files/353527/standalone.txt)
[standalone-full.txt](https://github.com/Comcast/cats/files/353528/standalone-full.txt)

Thanks.

**Comment by LTComcast** [comment]
1) Os version-  OS we are using is CentOS 7 ==> centos-release-7-2.1511
2) Java version- java version "1.7.0_79"
Java(TM) SE Runtime Environment (build 1.7.0_79-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.79-b02, mixed mode)
3) Mysql version - mysqladmin  Ver 8.42 Distrib 5.6.31, for Linux on x86_64
Server version		5.6.31
Protocol version	10
Connection		Localhost via UNIX socket
UNIX socket		/var/lib/mysql/mysql.sock
Uptime:			8 days 23 hours 21 min 23 sec
4) Mysql connector = mysql-connector-java-5.1.15
5) Attached both standalone files
[standalone-files.zip](https://github.com/Comcast/cats/files/353682/standalone-files.zip)

**Comment by Samruddhi123** [comment]
Any solutions from ur end ?

**Comment by XsubinX** [comment]
The JNDI lookup name system property is wrong

```
   <property name="keymanager.jndi.name" value="java:global/prod-cats-master-ear-1.0.1/keymanager-service/KeyManagerProxyImpl!com.comcast.cats.service.KeyManagerProxy"/>
```

As per the log,  you are trying to deploy **ocats-master-ear-1.1-OCATS-SNAPSHOT.ear**

The system property should be

```
<property name="keymanager.jndi.name" value="java:global/ocats-master-ear-1.1-OCATS-SNAPSHOT/keymanager-service/KeyManagerProxyImpl!com.comcast.cats.service.KeyManagerProxy"/>
```

**Comment by XsubinX** [comment]
Also make sure:
- you have keymanager database available in MySQL
- you have configured mysql driver in JBoss
- the username ans password in data source configuration is correct.

**Comment by LTComcast** [comment]
Hello Subin Sugunan,

Thanks for the help,

We have changed the property name as per your comments and also checked below configuration as well. Still we are getting same error. 


   1. keymanager database available in MySQL
   2.we have configured mysql driver in JBoss and added mysql connector version in module.xml file
   3. username ans password in data source configuration is correct.

Attached the jboss server log and standlalone xml file

Kindly let help us to resolve the issue
[deploy-error-logs.zip](https://github.com/Comcast/cats/files/358780/deploy-error-logs.zip)

**Comment by Samruddhi123** [comment]
Hello Subin Sugunan,
We have also changed the property name as per your comments and also checked above configuration guided by you BUT ==>  Still we are getting same error.

**Comment by XsubinX** [comment]
Looking at the standalone-full.xml. Clearly there is a conflict. JNDI name should match the version of EAR , which is getting deployed.


```
<property name="keymanager.jndi.name" value="java:global/ocats-master-ear-1.1-OCATS-SNAPSHOT/keymanager-service/KeyManagerProxyImpl!com.comcast.cats.service.KeyManagerProxy"/>
      <!--  <property name="keymanager.jndi.name" value="java:global/prod-cats-master-ear-1.0.1/keymanager-service/KeyManagerProxyImpl!com.comcast.cats.service.KeyManagerProxy"/>  -->
```


```
  <deployment name="prod-cats-master-ear-1.2.1.ear" runtime-name="prod-cats-master-ear-1.2.1.ear">
            <content sha1="2dab20adcb631a8ea938505a1c8a2ee93e8ce9b9"/>
        </deployment>
```
If you are deploying prod-cats-master-ear-1.2.1.ear , then use 

```
<property name="keymanager.jndi.name" value="java:global/ocats-master-ear-1.2.1/keymanager-service/KeyManagerProxyImpl!com.comcast.cats.service.KeyManagerProxy"/>
```

Thanks
Subin

**Comment by LTComcast** [comment]
Ear file name that we have used is "ocats-master-ear-1.1-OCATS-SNAPSHOT.ear". We have changed the deployment name accordingly. But still we are facing the same issue.

[logs-files-13-7-2016.zip](https://github.com/Comcast/cats/files/360900/logs-files-13-7-2016.zip)

**Comment by Samruddhi123** [comment]
Hello,
It seems that our MySQL JDBC driver: mysql-connector-java-5.1.39-bin.jar -> "com.mysql" is not correctly deployed.
We do not see our "com.mysql" driver in the "Create Datasource" screen while we try to add a datasource from JBOSS admin UI console. (screenshot: JBOSS_Create Datasource-screen.png)
We only see the default "h2" driver.

Also, in the JBOSS admin console UI, under Data sources, we only see the default "ExampleDS" data source. We do not see the pvr_db and keymanager datasources which we added in the standalone-full.xml (Screenshot: JBOSS_Datasources-screen.png)

Also, in the MySQL Driver directory where we have copied our "mysql-connector-java-5.1.39-bin".jar file, we do not see any "mysql-connector-java-5.1.39-bin".index file being generated Since, we installed/registered our jdbc connector as a Module as mentioned in our Install guide. (Screenshot: MySQL connector jar_Directory-screen.png)

Also, attaching the standalone-full.xml and module.xml for the same. (module_standalone-full files.zip

)



Can this be the root cause for our deployment of .war and .ear files?


Thanks,
![jboss_create datasource-screen](https://cloud.githubusercontent.com/assets/20223460/16803380/b1ddfcd8-4923-11e6-8583-3f0594225b8c.png)
![jboss_datasources-screen](https://cloud.githubusercontent.com/assets/20223460/16803381/b1ee4ce6-4923-11e6-82cd-9ee1ef951d13.png)

![mysql connector jar_directory-screen](https://cloud.githubusercontent.com/assets/20223460/16803382/b1f173c6-4923-11e6-8bb9-3090d318e41b.png)

**Comment by XsubinX** [comment]
I'm not sure the xml namespace you are using (**xmlns="urn:jboss:module:1.1"**) is correct. I think you must use **1.0** for JBoss 7.1.1.Final

Please follow below steps (I know you have already did most of it)

1) Create folder jboss-as-7.1.1.Final\modules\com\mysql\main
2) Download http://central.maven.org/maven2/mysql/mysql-connector-java/5.1.21/mysql-connector-java-5.1.21.jar  and copy it to jboss-as-7.1.1.Final\modules\com\mysql\main\mysql-connector-java-5.1.21.jar
3) Create file  jboss-as-7.1.1.Final\modules\com\mysql\main\module.xml and add below content

```
<?xml version="1.0" encoding="UTF-8"?>  
<module xmlns="urn:jboss:module:1.0" name="com.mysql">  
  <resources>  
    <resource-root path="mysql-connector-java-5.1.21.jar"/>  
  </resources>  
  <dependencies>  
    <module name="javax.api"/>  
  </dependencies>  
</module>
```

4) Add driver info in JBoss configuration file

Open jboss-as-7.1.1.Final\standalone\configuration\standalone.xml  or standalone-full.xml

and add

```
        <subsystem xmlns="urn:jboss:domain:datasources:1.0">
            <datasources>
                ...
                <drivers>
                    ...
			        <driver name="com.mysql" module="com.mysql">
                        <xa-datasource-class>com.mysql.jdbc.jdbc2.optional.MysqlXADataSource</xa-datasource-class>
                    </driver>
                </drivers>
            </datasources>
        </subsystem>
```

Restart JBoss

**Comment by XsubinX** [comment]
Now make sure keymanager data source is listed in admin console

![datasource-keymanager - config](https://cloud.githubusercontent.com/assets/808068/16810711/0b04d380-48f3-11e6-8004-803d3fb59f78.JPG)

**Comment by Samruddhi123** [comment]
Hello Subin,
Thank you for the detailed information. :)
We are now able to see Keymanager and pvr-db listed under Datasources in JBOSS Admin Console. (Screehshot: Datasources-List.png)
![datasources-list](https://cloud.githubusercontent.com/assets/20223460/16837167/337975d6-49df-11e6-87be-f556687d2f3c.png)


Also, we were able to deploy .WAR files: (Screenshot: Manage-deployment-screen.png)
1. **video-recorder-service-1.1-OCATS-SNAPSHOT.war**
2. **ocats-ui-1.1-OCATS-SNAPSHOT.war**

![manage-deployment-screen](https://cloud.githubusercontent.com/assets/20223460/16837176/400763f8-49df-11e6-8f5f-85a0970cec72.png)


**We are not able to deploy .EAR file : ocats-master-ear-1.1-OCATS-SNAPSHOT.ear**
We are facing a **new Error code** now while deploying the .EAR file

Attaching the Jboss server log and standalone.xml file (server.log_14-July.txt and standalone.xml)
[server.log_14-July.txt](https://github.com/Comcast/cats/files/363500/server.log_14-July.txt)
[standalone.xml.txt](https://github.com/Comcast/cats/files/363499/standalone.xml.txt)

**Error:**
`Unknown error`
`Unexpected HTTP response: 500`
`Request`
`{`
    `"address" => [("deployment" => "ocats-master-ear-1.1-OCATS-SNAPSHOT.ear")],`
`    "operation" => "deploy"`
`}`
`Response`
`Internal Server Error`
`{`
    `"outcome" => "failed",`
    `"failure-description" => {"**JBAS014671**: Failed services" => {"jboss.deployment.subunit.\"ocats-master-ear-1.1-OCATS-SNAPSHOT.ear\".\"ir-service.war\".component.KeyManagerImpl.START" => "org.jboss.msc.service.StartException in service jboss.deployment.subunit.\"ocats-master-ear-1.1-OCATS-SNAPSHOT.ear\".\"ir-service.war\".component.KeyManagerImpl.START: Failed to start service"}},`
    `"rolled-back" => true`
`}`

Thanks

**Comment by XsubinX** [comment]
## Connection refused error

```
Caused by: javax.ejb.EJBException: java.lang.RuntimeException: org.apache.http.conn.HttpHostConnectException: Connection to http://0.0.0.0 refused
```

Please try system property

```
<property name="cats.ir.proxy.ip" value="127.0.0.1:8080"/>
```

instead of 

```
<property name="cats.ir.proxy.ip" value="0.0.0.0"/>
```

## YAML errors

From https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc 

Get the following YAML files from /ocats-release/sampleConfigFiles

1.  rackconfig.catsrack
2   settopTypes.catsrack

copy the files in /var/www/html/prod-cats/config

```
cp *.catsrack /var/www/html/cats/config/
chown jboss:jboss –R /var/www/html/cats/config/
chmod 664 –R /var/www/html/cats/config/
```

Edit rackconfig.catsrack to reflect the rack configuration.

Also make sure you have configured this path as a system property

```
<property name="cats.config.path" value="/var/www/html/cats/config"/>
``` 

## Configuration file

The current OCATS is tested only using **standalone-full.xml.**  Recommended  use standalone-full.xml instead of standalone.xml to avoid any conflict

**Comment by Samruddhi123** [comment]
Hello Subin,
We have done the above mentioned steps/changes:
1. **Connection Refused error:**
<property name="cats.ir.proxy.ip" value="127.0.0.1:8080"/>

2. **YAML Errors:**
( Reference - https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc )
Get the following YAML files from /ocats-release/sampleConfigFiles
rackconfig.catsrack 2 settopTypes.catsrack
copy the files in /var/www/html/prod-cats/config

cp *.catsrack /var/www/html/cats/config/
chown jboss:jboss –R /var/www/html/cats/config/
chmod 664 –R /var/www/html/cats/config/

3. **System property already set to in standalone-full.xml:**
<property name="cats.config.path" value="/var/www/html/cats/config"/>

4. **Configuration file:**
We are now using the standalone-full.xml as default config file.

**With the above changes, we are now able to deploy the .EAR file : ocats-master-ear-1.1-OCATS-SNAPSHOT.ear**
**But, now we are NOT able to deploy the ocats-ui-1.1-OCATS-SNAPSHOT.war which was getting deployed before.** (screenshot: Manage-deployment-screen.png)
![manage-deployment-screen](https://cloud.githubusercontent.com/assets/20223460/16874158/61095224-4ab5-11e6-9198-15d69393ca77.png)

Facing the same Error code: (JBAS014671)

Attaching the standalone-full.xml and server log (server_15-July.log.txt and standalone-full.xml.txt)
[server_15-July.log.txt](https://github.com/Comcast/cats/files/365936/server_15-July.log.txt)
[standalone-full.xml.txt](https://github.com/Comcast/cats/files/365937/standalone-full.xml.txt)


Server logs causes:
17:47:20,539 ERROR [stderr] (MSC service thread 1-5) log4j:WARN No such property [maxFileSize] in org.apache.log4j.DailyRollingFileAppender.
17:47:20,539 ERROR [stderr] (MSC service thread 1-5) log4j:WARN No such property [maxBackupIndex] in org.apache.log4j.DailyRollingFileAppender.
17:47:20,561 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-5) MSC00001: Failed to start service jboss.deployment.unit."ocats-ui-1.1-OCATS-SNAPSHOT.war".component.UpTimeMonitor.START: org.jboss.msc.service.StartException in service jboss.deployment.unit."ocats-ui-1.1-OCATS-SNAPSHOT.war".component.UpTimeMonitor.START: Failed to start service
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1767) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [rt.jar:1.7.0_80]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [rt.jar:1.7.0_80]
	at java.lang.Thread.run(Thread.java:745) [rt.jar:1.7.0_80]
Caused by: java.lang.IllegalStateException: JBAS011048: Failed to construct component instance
	at org.jboss.as.ee.component.BasicComponent.constructComponentInstance(BasicComponent.java:163)
	at org.jboss.as.ee.component.BasicComponent.createInstance(BasicComponent.java:85)
	at org.jboss.as.ejb3.component.singleton.SingletonComponent.getComponentInstance(SingletonComponent.java:116)
	at org.jboss.as.ejb3.component.singleton.SingletonComponent.start(SingletonComponent.java:130)
	at org.jboss.as.ee.component.ComponentStartService.start(ComponentStartService.java:44)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1811) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1746) [jboss-msc-1.0.2.GA.jar:1.0.2.GA]
	... 3 more
Caused by: javax.ejb.EJBException: org.jboss.weld.exceptions.WeldException: WELD-000049 Unable to invoke [method] @PostConstruct public com.comcast.cats.config.ui.RackServiceImpl.init() on com.comcast.cats.config.ui.RackServiceImpl@2d3e77f4
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.handleExceptionInOurTx(CMTTxInterceptor.java:166)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:230)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.requiresNew(CMTTxInterceptor.java:333)
	at org.jboss.as.ejb3.tx.SingletonLifecycleCMTTxInterceptor.processInvocation(SingletonLifecycleCMTTxInterceptor.java:56)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ee.component.TCCLInterceptor.processInvocation(TCCLInterceptor.java:45)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:61) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ee.component.BasicComponent.constructComponentInstance(BasicComponent.java:161)
	... 9 more
Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049 Unable to invoke [method] @PostConstruct public com.comcast.cats.config.ui.RackServiceImpl.init() on com.comcast.cats.config.ui.RackServiceImpl@2d3e77f4
	at org.jboss.weld.bean.AbstractClassBean.defaultPostConstruct(AbstractClassBean.java:518)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget.postConstruct(ManagedBean.java:174)
	at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:291)
	at org.jboss.weld.context.AbstractContext.get(AbstractContext.java:107)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:616)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:681)
	at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:118)
	at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:703)
	at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:712)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget$1$1.proceed(ManagedBean.java:161)
	at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget$1.work(ManagedBean.java:157)
	at org.jboss.weld.bean.ManagedBean$FixInjectionPoint.run(ManagedBean.java:131)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget.inject(ManagedBean.java:153)
	at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:290)
	at org.jboss.weld.context.unbound.DependentContextImpl.get(DependentContextImpl.java:61)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:616)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:681)
	at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:118)
	at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:703)
	at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:712)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget$1$1.proceed(ManagedBean.java:161)
	at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget$1.work(ManagedBean.java:157)
	at org.jboss.weld.bean.ManagedBean$FixInjectionPoint.run(ManagedBean.java:131)
	at org.jboss.weld.bean.ManagedBean$ManagedBeanInjectionTarget.inject(ManagedBean.java:153)
	at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:290)
	at org.jboss.weld.context.AbstractContext.get(AbstractContext.java:107)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:616)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:681)
	at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:118)
	at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:703)
	at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:712)
	at org.jboss.weld.bean.SessionBean$1$1.proceed(SessionBean.java:176)
	at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
	at org.jboss.weld.bean.SessionBean$1.inject(SessionBean.java:173)
	at org.jboss.as.weld.injection.WeldEEInjection.inject(WeldEEInjection.java:78)
	at org.jboss.as.weld.injection.WeldInjectionInterceptor.processInvocation(WeldInjectionInterceptor.java:60)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ee.component.ManagedReferenceInterceptorFactory$ManagedReferenceInterceptor.processInvocation(ManagedReferenceInterceptorFactory.java:95)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.invocation.WeavedInterceptor.processInvocation(WeavedInterceptor.java:53) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:288) [jboss-invocation-1.1.1.Final.jar:1.1.1.Final]
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInOurTx(CMTTxInterceptor.java:228)
	... 18 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [rt.jar:1.7.0_80]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) [rt.jar:1.7.0_80]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) [rt.jar:1.7.0_80]
	at java.lang.reflect.Method.invoke(Method.java:606) [rt.jar:1.7.0_80]
	at org.jboss.weld.util.reflection.SecureReflections$13.work(SecureReflections.java:264)
	at org.jboss.weld.util.reflection.SecureReflectionAccess.run(SecureReflectionAccess.java:52)
	at org.jboss.weld.util.reflection.SecureReflectionAccess.runAsInvocation(SecureReflectionAccess.java:137)
	at org.jboss.weld.util.reflection.SecureReflections.invoke(SecureReflections.java:260)
	at org.jboss.weld.introspector.jlr.WeldMethodImpl.invoke(WeldMethodImpl.java:174)
	at org.jboss.weld.bean.AbstractClassBean.defaultPostConstruct(AbstractClassBean.java:516)
	... 63 more
Caused by: mapping values are not allowed here
 in "<reader>", line 2, column 5:
      ad: 244c7aaf-381d-4781-8319-c476ce ... 
        ^

	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchValue(ScannerImpl.java:864)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:358)
	at org.yaml.snakeyaml.scanner.ScannerImpl.peekToken(ScannerImpl.java:247)
	at org.yaml.snakeyaml.parser.ParserImpl$ParseDocumentEnd.produce(ParserImpl.java:264)
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:160)
	at org.yaml.snakeyaml.parser.ParserImpl.getEvent(ParserImpl.java:170)
	at org.yaml.snakeyaml.composer.Composer.composeDocument(Composer.java:124)
	at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:105)
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:120)
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:480)
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:411)
	at com.comcast.cats.service.util.YAMLUtils.loadFromYAML(YAMLUtils.java:62)
	at com.comcast.cats.config.ui.RackServiceImpl.readRacksFromDisk(RackServiceImpl.java:89)
	at com.comcast.cats.config.ui.RackServiceImpl.init(RackServiceImpl.java:79)
	... 73 more

17:47:20,782 INFO  [org.jboss.as.server] (HttpManagementService-threads - 3) JBAS015870: Deploy of deployment "ocats-ui-1.1-OCATS-SNAPSHOT.war" was rolled back with failure message {"JBAS014671: Failed services" => {"jboss.deployment.unit.\"ocats-ui-1.1-OCATS-SNAPSHOT.war\".component.UpTimeMonitor.START" => "org.jboss.msc.service.StartException in service jboss.deployment.unit.\"ocats-ui-1.1-OCATS-SNAPSHOT.war\".component.UpTimeMonitor.START: Failed to start service"}}
17:47:20,786 INFO  [org.jboss.as.webservices] (MSC service thread 1-6) JBAS015540: Stopping service jboss.ws.endpoint."ocats-ui-1.1-OCATS-SNAPSHOT.war".DeviceSearchServiceImpl
17:47:20,787 INFO  [org.jboss.ws.common.management.DefaultEndpointRegistry] (MSC service thread 1-6) remove: jboss.ws:context=CATS,endpoint=DeviceSearchServiceImpl
17:47:20,788 INFO  [org.jboss.as.webservices] (MSC service thread 1-7) JBAS015540: Stopping service jboss.ws.endpoint."ocats-ui-1.1-OCATS-SNAPSHOT.war".SettopAllocationServiceImpl
17:47:20,789 INFO  [org.jboss.ws.common.management.DefaultEndpointRegistry] (MSC service thread 1-7) remove: jboss.ws:context=CATS,endpoint=SettopAllocationServiceImpl
17:47:20,790 INFO  [org.jboss.weld.deployer] (MSC service thread 1-1) JBAS016009: Stopping weld service for deployment ocats-ui-1.1-OCATS-SNAPSHOT.war
17:47:20,831 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-7) JBAS015877: Stopped deployment ocats-ui-1.1-OCATS-SNAPSHOT.war in 48ms
17:47:20,832 INFO  [org.jboss.as.controller] (HttpManagementService-threads - 3) JBAS014774: Service status report
JBAS014777:   Services which failed to start:      service jboss.deployment.unit."ocats-ui-1.1-OCATS-SNAPSHOT.war".component.UpTimeMonitor.START: org.jboss.msc.service.StartException in service jboss.deployment.unit."ocats-ui-1.1-OCATS-SNAPSHOT.war".component.UpTimeMonitor.START: Failed to start service


Do you have any solutions for the above error?
Kindly let us know.
Thanks

**Comment by LTComcast** [comment]
Hello Subin,

Keymanager and Pvr datasources are not listed in the jboss console When we added  datasources  on Standalone-full.xml file . We added them in standalone.xml and after that we were able to see both the databases on jboss console

We have followed steps mentioned in this link and now also we are not able to deploy "ocats-master-ear-1.1-OCATS-SNAPSHOT.ear" and "video-recorder-service-1.1-OCATS-SNAPSHOT.war" file

while deploying video-recorder-service-1.1-OCATS-SNAPSHOT.war file it throws below error

[standalone@localhost:9999 /] deploy /root/.m2/repository/com/comcast/cats/video-recorder-service/1.1-OCATS-SNAPSHOT/video-recorder-service-1.1-OCATS-SNAPSHOT.war
{"JBAS014653: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-2" => "JBAS014750: Operation handler failed to complete"}}

attached both jboss server log and standalone.xml file .

Kindly let us know to resolve the error

[Log-standalone-15-7-2016.zip](https://github.com/Comcast/cats/files/366029/Log-standalone-15-7-2016.zip)

**Comment by XsubinX** [comment]
@LTComcast  there is no difference between the data source configurations in **standalone.xml** and **standalone-full.xml** . Please compere files and see what you are missing

**Comment by XsubinX** [comment]
## OCATS UI WAR Deployment

Not sure how you guys created **ocats-ui-1.1-OCATS-SNAPSHOT.war**

As per ocats-ui pom.xml ( https://github.com/Comcast/cats/blob/master/ocats-config/war/pom.xml )

_finalName_ is OCATS

```
<finalName>OCATS</finalName>
```

So once the build is complete, You must get  ocats-config\war\target\OCATS.war

**Before deploying , do the following**

copy below files 

- https://github.com/Comcast/cats/blob/master/ocats-release/sampleConfigFiles/rackconfig.catsrack
- https://github.com/Comcast/cats/blob/master/ocats-release/sampleConfigFiles/settopTypes.catsrack

 to the config folder as set in the system property

```
<property name="cats.config.path" value="/var/www/html/cats/config"/>
```

Set permissions

```
chmod 777 –R /var/www/html/cats/config/
```

Now deploy OCATS.war 

Once successfully deployed, access the ui at 

Access 

```
http://localhost:8080/CATS/
```

**Comment by XsubinX** [comment]
The deployments list page will looks like this

![ocats-ui-deployment](https://cloud.githubusercontent.com/assets/808068/16892436/d9ead674-4ae0-11e6-80b4-4e6d69dbcc5b.png)

**Comment by LTComcast** [comment]
Hello Subin,

Thanks a lot for your support and guidance

We are able to deploy all the war and ear files and able to see the OCCATS-ui, snmp-service,power-service, IR-service, Settop-service UI page

PFA for the screenshots of the same. 



![ir-services](https://cloud.githubusercontent.com/assets/19969537/16909807/871f391c-4cf2-11e6-9440-710e97a1c245.png)

**Comment by Samruddhi123** [comment]
Hello Subin,

We are still not able to deploy OCATS.war file and enable it. Please try and help us solve the error.
Please find the attached files. We are still facing the same above mentioned error.

[server18july.txt](https://github.com/Comcast/cats/files/368964/server18july.txt)
[standalone18july.txt](https://github.com/Comcast/cats/files/368963/standalone18july.txt)
[standalone-full18july.txt](https://github.com/Comcast/cats/files/368965/standalone-full18july.txt)

Thanks.

**Comment by XsubinX** [comment]
@LTComcast 

1) Regarding  http://127.0.0.1/video-recorder-service/rest/

Your JBoss is not fronted with Apache http server (httpd) and hence its accessible only at port 8080

Try

```
http://127.0.0.1:8080/video-recorder-service/rest/
```

2) Regarding  http://127.0.0.1/video-storage/

This is nothing but exposing a folder (/opt/video/) through http using Apache http server.

Refer **OCATS_INSTALL_GUIDE.doc**

Section "_CONFIGURE APACHE FOR VIDEO STORAGE_"

**Comment by Samruddhi123** [comment]
Hi Subin,

Thanks for your help. We are able to deploy all the 3 files and view OCATS UI. But we are still not able to view video-storage screen same like LTComcast.
Please find the attached server logs and screenshot for reference and help us debug the error.

![screenshot from 2016-07-19 17-15-11](https://cloud.githubusercontent.com/assets/20223460/16948968/7e7cc3cc-4dd5-11e6-99e9-7abaaacf95bb.png)
[server19july.txt](https://github.com/Comcast/cats/files/371370/server19july.txt)

**Comment by LTComcast** [comment]
Hello Subin,

We are getting " Forbidden - You don't have permission to access /video-storage/ on this server" error while accessing  http://127.0.0.1/video-storage/ link

Httpd server is running successfully and we are still 

[server.log.txt](https://github.com/Comcast/cats/files/371488/server.log.txt)

not able to view the http://127.0.0.1:8080/video-recorder-service/rest/ link  (Tried with http://127.0.0.1:8080/video-recorder-service-1.1-OCATS-SANPSHOT/rest/ also - throws an error)

[root@ocatsserver conf.d]# service httpd restart
Redirecting to /bin/systemctl restart  httpd.service
[root@ocatsserver conf.d]# service httpd service
The service command supports only basic LSB actions (start, stop, restart, try-restart, reload, force-reload, status). For other actions, please try to use systemctl.
[root@ocatsserver conf.d]# service httpd status
Redirecting to /bin/systemctl status  httpd.service
● httpd.service - The Apache HTTP Server
   Loaded: loaded (/etc/systemd/system/httpd.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2016-07-19 18:26:39 IST; 20s ago
     Docs: man:httpd(8)
           man:apachectl(8)
  Process: 27264 ExecStopPost=/usr/bin/kdestroy -A (code=exited, status=0/SUCCESS)
  Process: 27261 ExecStop=/bin/kill -WINCH ${MAINPID} (code=exited, status=1/FAILURE)
  Process: 27525 ExecStartPre=/usr/libexec/ipa/ipa-httpd-kdcproxy (code=exited, status=0/SUCCESS)
 Main PID: 27601 (httpd)
   Status: "Total requests: 0; Current requests/sec: 0; Current traffic:   0 B/sec"
   CGroup: /system.slice/httpd.service
           ├─27601 /usr/sbin/httpd -DFOREGROUND
           ├─27603 /usr/libexec/nss_pcache 3309582 off /etc/httpd/alias
           ├─27604 /usr/sbin/httpd -DFOREGROUND
           ├─27605 /usr/sbin/httpd -DFOREGROUND
           ├─27606 /usr/sbin/httpd -DFOREGROUND
           ├─27607 /usr/sbin/httpd -DFOREGROUND
           └─27608 /usr/sbin/httpd -DFOREGROUND

Jul 19 18:26:37 ocatsserver systemd[1]: Starting The Apache HTTP Server...
Jul 19 18:26:39 ocatsserver ipa-httpd-kdcproxy[27525]: ipa         : WARNING  Unable to connect to dirsrv: Timeout exceeded
Jul 19 18:26:39 ocatsserver ipa-httpd-kdcproxy[27525]: ipa         : WARNING  Disabling KDC proxy
**Jul 19 18:26:39 ocatsserver systemd[1]: Started The Apache HTTP Server.**

attaching jboss log for the reference . Kindly help us to proceed further

**Comment by LTComcast** [comment]
Hello Subin,

We are stuck up on accessing the  http://127.0.0.1/video-storage/ and  http://127.0.0.1:8080/video-recorder-service/rest/ files

As per the document we should have the httpd script file under  /etc/rc.d/init.d/ folder but as of now there is no httpd script file on that location (Reference - https://github.com/Comcast/cats/blob/master/docs/OCATS_INSTALL_GUIDE.doc) .
Kindly let us know whether  we should create a new file called httpd or will it be available by default?

**Installation Location:
/etc/httpd
Start-up script:
/etc/rc.d/init.d/httpd**

Also while accessing http://127.0.0.1/video-storage/ link we are getting error log file on /var/log/httpd/ folder

**[Wed Jul 20 16:58:26.664650 2016] [authz_core:error] [pid 23134] [client 127.0.0.1:34839] AH01630: client denied by server configuration: /opt/video/**

[error_log.txt](https://github.com/Comcast/cats/files/373594/error_log.txt)
attached the httpd error log file . Kindly help us to resolve the problem

Thanks!!

**Comment by XsubinX** [comment]
Regarding http://127.0.0.1/video-storage/

This is nothing but exposing a folder (E.g. /opt/video/) through http using Apache http server. CATS is doing nothing special here. Google how to serve static files using Apache http server. Make sure the folder exists and  it has write proper permission (777)

**Comment by XsubinX** [comment]
Regarding video-recorder-service

Rename _video-recorder-service-1.1-OCATS-SANPSHOT.war_ to **video-recorder-service.war** and deploy it. Now check if you can access UI at http://127.0.0.1:8080/video-recorder-service

**Comment by LTComcast** [comment]
Hello Subin,

after renaming the war file to video-recorder-service.war we are getting below error

type Exception report

message

description The server encountered an internal error () that prevented it from fulfilling this request.

exception

javax.servlet.ServletException: String index out of range: 0
	javax.faces.webapp.FacesServlet.service(FacesServlet.java:606)
	org.jboss.weld.servlet.ConversationPropagationFilter.doFilter(ConversationPropagationFilter.java:62)

root cause

java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	java.lang.String.charAt(String.java:658)
	com.sun.faces.application.view.MultiViewHandler.getResourceURL(MultiViewHandler.java:335)
	javax.faces.application.ViewHandlerWrapper.getResourceURL(ViewHandlerWrapper.java:244)
	org.jboss.weld.jsf.ConversationAwareViewHandler.getResourceURL(ConversationAwareViewHandler.java:141)
	org.primefaces.renderkit.CoreRenderer.getResourceURL(CoreRenderer.java:73)
	org.primefaces.renderkit.OutcomeTargetRenderer.getTargetURL(OutcomeTargetRenderer.java:91)
	org.primefaces.component.menu.BaseMenuRenderer.encodeMenuItem(BaseMenuRenderer.java:77)
	org.primefaces.component.tieredmenu.TieredMenuRenderer.encodeMenuContent(TieredMenuRenderer.java:96)
	org.primefaces.component.tieredmenu.TieredMenuRenderer.encodeSubmenu(TieredMenuRenderer.java:155)
	org.primefaces.component.tieredmenu.TieredMenuRenderer.encodeMenuContent(TieredMenuRenderer.java:112)
	org.primefaces.component.tieredmenu.TieredMenuRenderer.encodeMenu(TieredMenuRenderer.java:77)
	org.primefaces.component.menubar.MenubarRenderer.encodeMarkup(MenubarRenderer.java:54)
	org.primefaces.component.menu.BaseMenuRenderer.encodeEnd(BaseMenuRenderer.java:38)
	javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:875)
	com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:312)
	com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:105)
	javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:845)
	javax.faces.component.UIComponent.encodeAll(UIComponent.java:1779)
	javax.faces.render.Renderer.encodeChildren(Renderer.java:168)
	javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:845)
	com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:304)
	com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:105)
	javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:845)
	javax.faces.component.UIComponent.encodeAll(UIComponent.java:1779)
	javax.faces.component.UIComponent.encodeAll(UIComponent.java:1782)
	javax.faces.component.UIComponent.encodeAll(UIComponent.java:1782)
	com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:402)
	com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:125)
	javax.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:288)
	com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:121)
	com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
	com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:139)
	javax.faces.webapp.FacesServlet.service(FacesServlet.java:594)
	org.jboss.weld.servlet.ConversationPropagationFilter.doFilter(ConversationPropagationFilter.java:62)
[https-log.txt](https://github.com/Comcast/cats/files/373716/https-log.txt)
[server.log.txt](https://github.com/Comcast/cats/files/373717/server.log.txt)

**Comment by XsubinX** [comment]
@LTComcast 

I was able to reproduce and fix the error. Created a separate issue  #30 to track the same . 

I will create a pull request separately. 

 In the mean time, You can build your own war file referring the fix description in #30 

Thanks
Subin

**Comment by gouthmi** [comment]
Hi,

I am facing issue in DataSources and also not able to deploy the war files.

1.In Datasources of JBOSS console UI , it is showing enabled true for all the datasources listed there.(attached datasource.png).
But  when clicking on Datasouces-->keymanager, I am getting error (Attached screenshot --> keymanager_error.png).
The same is happening for pvr_db also.

2.I am successfully able to deploy the OCATS war file.But unable to deploy ocats-master-ear-1.1-OCATS-SNAPSHOT.ear and video-recorder-service.war.
Attached screenshots --> OCATS_EAR_Error.png , Deployement.png and video-services.png


Also attaching the standalone-full.xml.


Could you please help.
![datasources](https://cloud.githubusercontent.com/assets/26746999/24418292/69b59066-13e3-11e7-90af-dfed97937ff7.png)
![deployement](https://cloud.githubusercontent.com/assets/26746999/24418291/69b4d752-13e3-11e7-8cee-41d0f16dec70.png)
![keymanager_error](https://cloud.githubusercontent.com/assets/26746999/24418289/69b2a734-13e3-11e7-8fb1-01a31d6962d9.png)
![ocats_ear_error](https://cloud.githubusercontent.com/assets/26746999/24418290/69b335fa-13e3-11e7-95b7-614a0e8c49f0.png)
![video-services](https://cloud.githubusercontent.com/assets/26746999/24418288/69b267d8-13e3-11e7-917e-d2e1a73e3dc7.png)

**Comment by gouthmi** [comment]
[standalone-full.txt](https://github.com/Comcast/cats/files/876505/standalone-full.txt)

**Comment by sjsauto** [comment]
Is the apache setup done ?

**Comment by gouthmi** [comment]
Yea..Apache is running.

Please find the below logs:

[centos@centos etc]$ service httpd status
Redirecting to /bin/systemctl status  httpd.service
● httpd.service - The Apache HTTP Server
   Loaded: loaded (/usr/lib/systemd/system/httpd.service; disabled; vendor preset: disabled)
   Active: active (running) since Tue 2017-03-28 15:21:43 BST; 17h ago
     Docs: man:httpd(8)
           man:apachectl(8)
 Main PID: 4569 (httpd)
   Status: "Total requests: 0; Current requests/sec: 0; Current traffic:   0 B/sec"
   CGroup: /system.slice/httpd.service
           ├─4569 /usr/sbin/httpd -DFOREGROUND
           ├─4573 /usr/sbin/httpd -DFOREGROUND
           ├─4574 /usr/sbin/httpd -DFOREGROUND
           ├─4575 /usr/sbin/httpd -DFOREGROUND
           ├─4577 /usr/sbin/httpd -DFOREGROUND
           └─4578 /usr/sbin/httpd -DFOREGROUND
